lib/spack/llnl/util/filesystem.py

```diff
@@ -1472,7 +1472,7 @@ def set_executable(path):
 def recursive_mtime_greater_than(path: str, time: float) -> bool:
     """Returns true if any file or dir recursively under `path` has mtime greater than `time`."""
     # use bfs order to increase likelihood of early return
-    queue: Deque[str] = collections.deque()
+    queue: Deque[str] = collections.deque(path)
```

Comment:
I think this should be
``` suggestion
    queue: Deque[str] = collections.deque([path])
```

Reply:
We have type checking enabled as part of CI via the `typing` module so CI should have caught this if it was wrong. Example of being caught wrong here: https://github.com/spack/spack/actions/runs/12959094151/job/36150747225.

The brackets in this case are to allow you to specify a list of acceptable types:. See some examples here: https://docs.python.org/3/library/typing.html#type-aliases

Reply:
oh hmm I did not mean as a type problem, I meant that I think it is a problem with the initialization of `queue`. It looks like if `path = "/path/to/file"` then `queue = ['p', 'a', 't', 'h', '/', 't', 'o', '/', 'f', 'i', 'l', 'e']` instead of `queue = ["/path/to/file"]`

Reply:
Interesting. I tested it locally prior to merging to check if this failed because I had this concern too. Apologies if it was still broken. Should have added a test.

Reply:
No problem! Thanks for the quick response. Think all is working now as far as I can tell!